nav2_waypoint_follower/include/nav2_waypoint_follower/waypoint_follower.hpp

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2019 Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_WAYPOINT_FOLLOWER__WAYPOINT_FOLLOWER_HPP_
+#define NAV2_WAYPOINT_FOLLOWER__WAYPOINT_FOLLOWER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_msgs/action/navigate_to_pose.hpp"
+#include "nav2_msgs/action/follow_waypoints.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/simple_action_server.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+
+namespace nav2_waypoint_follower
+{
+
+enum class ActionStatus
+{
+  UNKNOWN = 0,
+  PROCESSING = 1,
+  FAILED = 2,
+  SUCCEEDED = 3
+};
+
+/**
+ * @class nav2_waypoint_follower::WaypointFollower
+ * @brief An action server that uses behavior tree for navigating a robot to its
+ * goal position.
+ */
+class WaypointFollower : public nav2_util::LifecycleNode
+{
+public:
+  using ActionT = nav2_msgs::action::FollowWaypoints;
+  using ClientT = nav2_msgs::action::NavigateToPose;
+  using ActionServer = nav2_util::SimpleActionServer<ActionT>;
+  using ActionClient = rclcpp_action::Client<nav2_msgs::action::NavigateToPose>;
```

Comment:
using ActionClient = rclcpp_action::Client<ClientT>;

Reply:
Why?

Reply:
Oops, still learning how to use vscode to do code reviews, for some reason, it displays differently on the UI. Maybe vscode parses `<>` as a comment.

Anyhow, just suggesting a minor simplification: `using ActionClient = rclcpp_action::Client<ClientT>;`



![image](https://user-images.githubusercontent.com/39749557/69985622-6bb5bb00-14f0-11ea-8f09-ac89a1d2ada2.png)

Reply:
Oh yeah, sure! I'll make that change.